projects/swimlane/ngx-graph/src/lib/graph/graph.component.ts

```diff
@@ -508,6 +518,63 @@ export class GraphComponent extends BaseChartComponent implements OnInit, OnChan
     this.cd.markForCheck();
   }
 
+  getMinimapTransform(): string {
+    switch (this.miniMapPosition) {
+      case MiniMapPosition.UpperLeft: {
+        return '';
+      }
+      case MiniMapPosition.UpperRight: {
+        return 'translate(' + (this.dims.width - this.graphDims.width / this.minimapScaleCoefficient) + ',' + 0 + ')';
+      }
+      default: {
+        return '';
+      }
+    }
+  }
+
+  updateGraphDims() {
+    let minX = +Infinity;
+    let maxX = -Infinity;
+    let minY = +Infinity;
+    let maxY = -Infinity;
+
+    for (let i = 0; i < this.graph.nodes.length; i++) {
+      const node = this.graph.nodes[i];
+      minX = node.position.x < minX ? node.position.x : minX;
+      minY = node.position.y < minY ? node.position.y : minY;
+      maxX = node.position.x + node.dimension.width > maxX ? node.position.x + node.dimension.width : maxX;
+      maxY = node.position.y + node.dimension.height > maxY ? node.position.y + node.dimension.height : maxY;
+    }
+    minX -= 100;
```

Comment:
why 100? consider extract to a const value because it repeats

Reply:
This is to add some padding around the nodes on the minimap, so they do not show on the edge. We can play with this value. Also good point, I extracted it into a const.